sales/src/execute.rs

```diff
@@ -0,0 +1,209 @@
+                        to_address: ownable
+                            .borrow()
+                            .get_owner(&deps.as_ref())
+                            .unwrap()
+                            .to_string(),
+                        amount: vec![Coin::new(
+                            paying_fund.amount.u128(),
+                            paying_fund.denom.clone(),
+                        )],
+                    };
+                    let cosmos_msg = CosmosMsg::Bank(message);
+                    response = response.add_message(cosmos_msg);
+
+                    if paying_fund.amount.lt(&info.funds[0].amount) {
+                        // refund user back extra funds
+                        let refund_amount = info.funds[0]
+                            .amount
+                            .checked_sub(paying_fund.amount)
+                            .unwrap();
+                        let refund_message = BankMsg::Send {
+                            to_address: info.sender.to_string(),
+                            amount: vec![Coin::new(
+                                refund_amount.u128(),
+                                paying_fund.denom.clone(),
+                            )],
+                        };
+                        let refund_cosmos_msg = CosmosMsg::Bank(refund_message);
+                        response = response.add_message(refund_cosmos_msg);
+                    }
+                    self.primary_sales.save(deps.storage, &primary_sales)?;
+                    return Ok(response);
+                }
+            }
+        }
+        return Err(ContractError::NoOngoingPrimarySaleError);
+    }
+
+    pub fn mint(
+        &self,
+        deps: &mut DepsMut,
+        _env: Env,
+        info: &MessageInfo,
+        msg: MintMsg<T>,
+    ) -> Result<Response, ContractError> {
+        // create the token
+        let token = TokenInfo {
+            owner: deps.api.addr_validate(&msg.owner)?,
+            approvals: vec![],
+            token_uri: msg.token_uri,
+            extension: msg.extension,
+        };
+        {
+            self.sellable
+                .borrow()
+                .tokens
+                .borrow_mut()
+                .contract
+                .tokens
+                .update(deps.storage, &msg.token_id, |old| match old {
+                    Some(_) => Err(ContractError::TokenModuleError(
+                        cw721_base::ContractError::Claimed {},
+                    )),
+                    None => Ok(token),
+                })?;
+        }
+        self.sellable
+            .borrow()
+            .tokens
+            .borrow_mut()
+            .contract
+            .increment_tokens(deps.storage)?;
+
+        Ok(Response::new()
+            .add_attribute("action", "mint")
+            .add_attribute("minter", &info.sender)
+            .add_attribute("owner", msg.owner)
+            .add_attribute("token_id", msg.token_id))
+    }
+}
+
+fn check_ownable(
```

Comment:
Love this helper. Let's call this `assert_owner`.

Reply:
Done